File: ThreadsApp/ThreadsApp/View/Chat/Cells/Button/ButtonCell.m

//
//  ButtonCell.m
//  ThreadsApp
//
//  Created by Tabriz Dzhavadov on 13/07/2017.
//  Copyright © 2017 Brooma Service. All rights reserved.
//

#import "ButtonCell.h"

@interface ButtonCell()

@end

@implementation ButtonCell

- (void) awakeFromNib {
    [super awakeFromNib];
    self.selectionStyle = UITableViewCellSelectionStyleNone;
}

- (void) construct {
    [self.button setTitle: [CellTypeHelper textByType: self.type] forState: UIControlStateNormal];
}

- (IBAction) clicked: (UIButton *) sender {
    if (self.delegate && [self.delegate respondsToSelector: @selector(buttonCellClicked:)]) {
        [self.delegate buttonCellClicked: self];
    }
}

@end

File: ThreadsApp/ThreadsAppSwift/View/Chat/Cells/ButtonCell.swift
//
//  ButtonCell.swift
//  THRAppSwift-Polarbear
//
//  Created by Brooma Service on 16/05/2019.
//  Copyright © 2019 Brooma Service. All rights reserved.
//

import Foundation
import UIKit

protocol ButtonCellDelegate: class {
    func buttonCellClicked(_ cell: ButtonCell)
}

class ButtonCell : UITableViewCell {
    
    var type: CellType = .FullChat
    
    weak var delegate: ButtonCellDelegate?
    
    @IBOutlet weak var button: UIButton!
    
    override func awakeFromNib() {
        super.awakeFromNib()
        self.selectionStyle = .none
    }
    
    func construct() {
        self.button.setTitle(CellTypeHelper.getCellText(forType: self.type), for: .normal)
    }
    
    @IBAction func clicked(_ sender: Any) {    
        delegate?.buttonCellClicked(self)
    }
    
}
